### PR TITLE
feat(macOS): Window sizing

### LINKED ISSUE #3435: macOS Window sizing
## What would you like to be added:

Set window size on macOS via `ApplicationView`

## Why is this needed:

UWP parity.

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #3435 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

Not supported.

## What is the new behavior?

- `ApplicationView.TryResizeView` and `SetPreferredMinSize` on macOS
- `CoreWindow.SizeChanged` on all platforms (forwarded from `Window.SizeChanged`)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UWP/UI/ViewManagement/ApplicationView.macOS.cs [hunk 1]
```diff
@@ -64,6 +66,24 @@ public void ExitFullScreenMode()
...
+			var window = NSApplication.SharedApplication.KeyWindow;
+			var frame = window.Frame;
+			frame.Size = new CGSize(value.Width, value.Height);
+			window.SetFrame(frame, true, true);
+			return true;
+		}
+
+		public void SetPreferredMinSize(Size minSize)
+		{
+			var window = NSApplication.SharedApplication.KeyWindow;
```
Comment (davidjohnoliver) [inline src/Uno.UWP/UI/ViewManagement/ApplicationView.macOS.cs:83]: Should `VerifyKeyWindowInitialized()` also be called here?

File: src/Uno.UWP/UI/ViewManagement/ApplicationView.macOS.cs [hunk 2]
```diff
@@ -64,6 +66,24 @@ public void ExitFullScreenMode()
...
 		}
 
+
+		public bool TryResizeView(Size value)
+		{
+			VerifyKeyWindowInitialized();
+
+			var window = NSApplication.SharedApplication.KeyWindow;
+			var frame = window.Frame;
+			frame.Size = new CGSize(value.Width, value.Height);
```
Comment (davidjohnoliver) [inline src/Uno.UWP/UI/ViewManagement/ApplicationView.macOS.cs:76]: there's an implicit conversion between CGSize/Size I believe

File: src/Uno.UWP/UI/ViewManagement/ApplicationView.macOS.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UWP/UI/ViewManagement/ApplicationView.macOS.cs:83]: Right!

**Comment by MartinZikmund** [comment]
@jeromelaban The build is failing on WinUI - cannot convert from 'Microsoft.UI.Xaml.WindowSizeChangedEventArgs' to 'Windows.UI.Core.WindowSizeChangedEventArgs'. How do I go about fixing that please?

**Comment by jeromelaban** [comment]
@MartinZikmund this is generally caused by a missing using namespace in the modified file. A reliable modification is to qualify explicitly this type in all locations.

**Pull request merged**